presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -13,6 +13,8 @@
  */
```

Comment:
WRT to commit message. It looks like the refactor is not the most important change here. The most important part is that you added missing invalidate for caches. And refactor is only about preventing this to happen in future again. Please consider writing better commit message.

Reply:
couldn't find a better message: it's not just a bugfix, but also refactoring to remove duplication

Reply:
> it's not just a bugfix, but also refactoring to remove duplication

So maybe two commits. One bugfix. Second refactofing. WDYT?